autoscale/src/main/java/com/sequenceiq/periscope/service/YarnBasedScalingAdjustmentService.java

```diff
@@ -130,19 +150,26 @@ protected boolean isCoolDownTimeElapsed(String clusterCrn, String coolDownAction
     }
 
     private void publishScalingEventIfNeeded(Cluster cluster, int finalScaleUpCount, StackV4Response stackV4Response, List<String> hostsToDecommission,
-            ScalingAdjustmentType adjustmentType) {
+            ScalingAdjustmentType adjustmentType, String yarnRecommendationMessage) {
+        StringBuilder sb = new StringBuilder(yarnRecommendationMessage);
         if (upscalable(cluster, finalScaleUpCount))  {
+            ScalingActivity scalingActivity =
+                    scalingActivityService.create(cluster, METRICS_COLLECTION_SUCCESS, sb.append(UPSCALE_MESSAGE).toString(), clock.getCurrentTimeMillis());
```

Comment:
Don't see METRICS_COLLECTION_FAILED being set anywhere ?

Reply:
Updated to set `METRICS_COLLECTION_FAILED` in `UpdateFailedHandler` if multiple exceptions wrt to YARN polling are encountered